### ISSUE TITLE
Content disappears using Dialog with Adapt on strict mode

### ISSUE THREAD

**Issue by carlpontus** [tamagui/tamagui#2944]
### Current Behavior

The content within `<Dialog.Content>` disappears when using Dialog with Adapt on strict mode. The behaviour appears on web when adjusting screen size on strict mode.

It works if removing strict mode.

https://github.com/user-attachments/assets/d4e48011-9cd8-48ac-abff-c489b489c4db

### Expected Behavior

Content being persisted when adjusting screen size using dialog with Adapt even in strict mode

### Tamagui Version

```markdown
1.112.0
```

### Platform (Web, iOS, Android)

```markdown
Web
```

### Reproduction

```markdown
1. Run npm create tamagui@latest
2. Select `Learn - Vite + Webpack, Tamagui config from scratch`
3. Swap out `<Root />` for `<React.StrictMode><Root /></React.StrictMode>` in index.ts
4. Add a dialog with adapt in Root.tsx like the following:
```
```typescript
    <YStack>
        <Dialog modal>
          <Dialog.Trigger asChild>
            <Button>Show Dialog</Button>
          </Dialog.Trigger>
          <Adapt when="sm">
            <Sheet animation="quick" zIndex={200000} modal dismissOnSnapToBottom>
              <Sheet.Frame padding="$4" gap="$4">
                <Adapt.Contents />
              </Sheet.Frame>
              <Sheet.Overlay
                animation="lazy"
                enterStyle={{ opacity: 0 }}
                exitStyle={{ opacity: 0 }}
              />
            </Sheet>
          </Adapt>
          <Dialog.Portal>
            <Dialog.Overlay
              key="overlay"
              animation="quick"
              opacity={0.5}
              enterStyle={{ opacity: 0 }}
              exitStyle={{ opacity: 0 }}
            />
            <Dialog.Content
              bordered
              elevate
              key="content"
              animateOnly={['transform', 'opacity']}
              animation={[
                'quick',
                {
                  opacity: {
                    overshootClamping: true,
                  },
                },
              ]}
              enterStyle={{ x: 0, y: -20, opacity: 0, scale: 0.9 }}
              exitStyle={{ x: 0, y: 10, opacity: 0, scale: 0.95 }}
              gap="$4"
            >
              <Dialog.Title>DIALOG</Dialog.Title>
            </Dialog.Content>
          </Dialog.Portal>
        </Dialog>
      </YStack>
```

### System Info

```markdown
System:
    OS: macOS 14.6.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 331.42 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v20.10.0/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    pnpm: 9.11.0 - ~/.nvm/versions/node/v20.10.0/bin/pnpm
  Browsers:
    Chrome: 129.0.6668.60
    Safari: 17.6
  npmPackages:
    @pmmmwh/react-refresh-webpack-plugin: ^0.5.11 => 0.5.15 
    @tamagui/animations-css: ^1.112.0 => 1.112.0 
    @tamagui/core: ^1.112.0 => 1.112.0 
    @tamagui/create-theme: ^1.112.0 => 1.112.0 
    @tamagui/react-native-svg: ^1.112.0 => 1.112.0 
    @tamagui/vite-plugin: ^1.112.0 => 1.112.0 
    @vitejs/plugin-react-swc: ^3.6.0 => 3.7.0 
    async-retry: 1.3.3 => 1.3.3 
    babel-polyfill: ^6.26.0 => 6.26.0 
    css-loader: ^6.10.0 => 6.11.0 
    esbuild-loader: ^4.1.0 => 4.2.2 
    esbuild-register: ^3.5.0 => 3.6.0 
    expo-linear-gradient: ^12.7.2 => 12.7.2 
    file-loader: ^6.2.0 => 6.2.0 
    html-webpack-plugin: ^5.6.0 => 5.6.0 
    mini-css-extract-plugin: ^2.8.1 => 2.9.1 
    react: ^18.2.0 || ^19.0.0 => 18.3.1 
    react-dom: ^18.2.0 || ^19.0.0 => 18.3.1 
    react-native-web: ^0.19.10 => 0.19.12 
    react-refresh: ^0.14.0 => 0.14.2 
    style-loader: ^3.3.4 => 3.3.4 
    tamagui: ^1.112.0 => 1.112.0 
    tamagui-loader: ^1.112.0 => 1.112.0 
    thread-loader: ^4.0.2 => 4.0.4 
    url-loader: ^4.1.1 => 4.1.1 
    vite: 6.0.0-beta.1 => 6.0.0-beta.1 
    webpack: ^5.91.0 => 5.94.0 
    webpack-bundle-analyzer: ^4.10.1 => 4.10.2 
    webpack-cli: ^5.1.4 => 5.1.4 
    webpack-dev-server: ^5.0.4 => 5.1.0
```

**Comment by ScriptShah** [comment]
I like work on this please assign me.

**Comment by natew** [comment]
@carlpontus can you verify this is still breaking on the latest, we did some media query correctness work

**Comment by carlpontus** [comment]
It is fixed now, tested on v1.116.7 @natew!
